plugins/pagerduty/config.d.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export interface Config {
+  /**
+   * Configuration for the PagerDuty plugin
+   * @visibility frontend
+   */
+  pagerduty?: {
```

Comment:
```suggestion
  pagerDuty?: {
```

seems to be the format! https://github.com/backstage/backstage/blob/55d0350ab5daf16d14bdbae0f258bba6fcc96d45/plugins/pagerduty/src/api/client.ts#L55

Which begs the question ... who's being case insensitive, and is it the right thing to be? 🤔

Reply:
That's a great point. We had a similar where we had to change `sonarqube` to `sonarQube` due to the `--strict` requirement, which is case sensitive.

Seems the config package in the getOptionalString() method just brings in the value, but isn't checking the value so far as I can see.  That infers to your point that the config reader basically doesn't care; it's maybe only the CLI `--strict` flag  which does?

So maybe the CLI has enforced it TOO strictly and that should be made case Insensitive so we don't hit these issues?

Otherwise if I release this, it's not TECHNICALLY a breaking change. But if someone pulls the package in, and like we have in our internal pipeline, it'll fail with the `--strict` flag requiring the user to fix the case in our `app-config.yaml` files.

Reply:
Let me know if you think I should call that out in the changeset. It's likely a trivial point, and isn't breaking PER se, it's just being a bit more.... explicit I suppose, and is generally an easy "fix" of just fixing the case in a config YAML.  Welcome for counter opinions!

Reply:
Note other context re: the case sensitivity, with my team submitting some other PRs for other relevant case issues we ran into: https://github.com/backstage/backstage/pull/18239

Reply:
Just tested this locally, with the current `app-config.yaml` running `getOptionalString('pagerDuty.eventsBaseUrl')` gives the config value and `getOptionalString('pagerduty.eventsBaseUrl')` gives `undefined`. I think we're hitting the null coalescing from the linked section, `?? 'https://events.pagerduty.com/v2'` which is the same as the config value and hence why this wasn't more clear.

Both the CLI and config reader are case sensitive, I think this is just a case where the config value and default value were the same.

Reply:
@sennyeya yep, when we put in `sonarQube` now our strict run is clean. i even just tried putting it back to lower, and it (properly?) fails again:

<img width="1181" alt="image" src="https://github.com/backstage/backstage/assets/33203301/bf317c2c-7848-4714-baf0-697b032908af">

Reply:
Sorry, I'd meant on the plugin side, is your SonarQube plugin working correctly? Similarly with PagerDuty after this change, especially on custom/self-hosted instances, if you can test with one.

Reply:
Yep good call; looks like my colleague @ritesh007 may be firing a fix for that separately!